### PR TITLE
Sites-List Removal: Sharing Controller

### PR DESCRIPTION
**Pull request by lamosty**
In this PR, we are removing sites-list from `my-sites/sharing/controller` file.

## Testing

Navigate to `/sharing/<site>` and clear app caches. Reload the page and try playing with it – working as before?

### CODE REVIEW

File: client/my-sites/sharing/controller.js [hunk 1]
```diff
@@ -18,17 +18,25 @@ import { sectionify } from 'lib/route';
...
 	const { contentComponent, path, store } = context;
+	const state = store.getState();
 
-	if ( site && ! site.settings && utils.userCan( 'manage_options', site ) ) {
-		site.fetchSettings();
+	const siteId = getSelectedSiteId( state );
+	const siteSettings = getSiteSettings( state, siteId );
+
+	if ( siteId && ! siteSettings && canCurrentUser( state, siteId, 'manage_options' ) ) {
+		reduxDispatch( requestSiteSettings( siteId ) );
```
Comment (jsnajdr) [inline client/my-sites/sharing/controller.js]: You shouldn't need to use the `redux-bridge` in a `page` controller like this: you can directly call `store.dispatch( ... )`.

The bridge is only for modules that can't access the store in any other way, i.e., Flux stores.

Comment (lamosty) [reply client/my-sites/sharing/controller.js]: Ah, didn't know. Thanks!

File: client/my-sites/sharing/controller.js [hunk 2]
```diff
@@ -18,17 +18,24 @@ import { sectionify } from 'lib/route';
...
 	const { contentComponent, path, store } = context;
+	const state = store.getState();
 
-	if ( site && ! site.settings && utils.userCan( 'manage_options', site ) ) {
-		site.fetchSettings();
+	const siteId = getSelectedSiteId( state );
+	const siteSettings = getSiteSettings( state, siteId );
+
+	if ( siteId && ! siteSettings && canCurrentUser( state, siteId, 'manage_options' ) ) {
+		store.dispatch( requestSiteSettings( siteId ) );
```
Comment (jsnajdr) [inline client/my-sites/sharing/controller.js]: Issuing requests for data that are needed to render the UI is usually done inside the React components that need them, by rendering `<QuerySiteSettings />`. Is the `requestSiteSettings` really still needed here? The `sharing/buttons/buttons` component uses `QuerySiteSettings` and other component don't seem to use the site settings data.

Comment (lamosty) [reply client/my-sites/sharing/controller.js]: That's a good point. I'll remove this code.

**Pull request merged**